### PR TITLE
Coverage v3

### PR DESCRIPTION
**Pull request by nikfilippas**
In `v3_wip` we accidentally decreased the overall coverage a little.
This PR includes testing for all new features we introduced.
The rest is deprecated stuff which we will not be testing.

### CODE REVIEW

File: pyccl/base/parameters/parameters_base.py [hunk 1]
```diff
@@ -74,6 +74,10 @@ def _new_setattr(self, key, value):
 
     def __init__(self):
         # Create a new instance if a factory is provided.
+        if not (hasattr(self, "_instance") or hasattr(self, "_factory")):
+            name = type(self).__name__
+            raise TypeError(f"Can't instantiate {name} with no set "
+                            "`instance` or `factory`.")
```
Comment (damonge) [inline pyccl/base/parameters/parameters_base.py:80]: Please explain what you did in this file and why

File: pyccl/tests/test_power.py [hunk 2]
```diff
@@ -257,6 +257,14 @@ def test_input_lin_power_spectrum():
...
     assert np.allclose(pk_CCL_input, pk_CCL, atol=0., rtol=1e-5)
 
+    with pytest.raises(ValueError):
+        ccl.CosmologyCalculator(
+            Omega_c=0.27, Omega_b=0.05, h=0.7,
+            n_s=0.965, A_s=2e-9,
+            background={'a': a_arr,
+                        'chi': chi_from_ccl,
+                        'h_over_h0': hoh0_from_ccl[:-1]})
+
```
Comment (damonge) [inline pyccl/tests/test_power.py:267]: Can you comment on what this tests?

File: pyccl/base/parameters/parameters_base.py (hunk 1 above)
Comment (nikfilippas) [reply pyccl/base/parameters/parameters_base.py:80]: done

File: pyccl/tests/test_power.py (hunk 2 above)
Comment (nikfilippas) [reply pyccl/tests/test_power.py:268]: done

**Comment by nikfilippas** [comment]
@damonge ok I think we're done now

File: pyccl/base/parameters/parameters_base.py (hunk 1 above)
Comment (damonge) [reply pyccl/base/parameters/parameters_base.py:80]: No, I mean explain this to me. No idea what the comment means

Comment (nikfilippas) [reply pyccl/base/parameters/parameters_base.py:80]: prevents `CCLParameters` from getting instantiated

**Review by damonge — APPROVED** [review]
LGTM

**Pull request merged**